R/checkout.R

```diff
@@ -2,21 +2,48 @@
 #' Checkout a Repository
 #'
 #' `renv::checkout()` can be used to install the latest packages available from
-#' the requested repositories.
+#' a (set of) package repositories.
+#'
+#' `renv::checkout()` is most useful with services like the Posit's
+#' [Package Manager](https://packagemanager.rstudio.com/), as it
+#' can be used to switch between different repository snapshots within an
+#' `renv` project. In this way, you can upgrade (or downgrade) all of the
+#' packages used in a particular `renv` project to the package versions
+#' provided by a particular snapshot.
```

Comment:
Mention that it will also convert dev versions installed from Github to released versions? Or maybe in general that it matches the names of currently used packages to available packages from the specified repos?